src/components/PlayerSubmissionForm.js

```diff
@@ -1,23 +1,120 @@
 import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-
+import Game from './Game';
 import './PlayerSubmissionForm.css';
 
-const PlayerSubmissionForm = () => {
+const PlayerSubmissionForm = ({sendSubmission, index, fields}) => {
+  const [inputFields, setInputFields] = useState({
+    adj1: '',
+    noun1: '',
+    adv: '',
+    verb: '',
+    adj2: '',
+    noun2: '',
+  });
+
+  const onInputChange = (event) => {
+    const {name, value} = event.target;
+
+    const newFormFieldValues = {
+      ...inputFields,
+    }
+
+    newFormFieldValues[name] = value;
+    setInputFields(newFormFieldValues);
+  };
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    sendSubmission(inputFields);
+
+    setInputFields({
+      adj1: '',
+      noun1: '',
+      adv: '',
+      verb: '',
+      adj2: '',
+      noun2: '',
+    });
+  }
+
+  const poemFields = fields.map((field, i) => {
+    if (typeof field === 'string') {
+      return (
+        <div key={i}>
+          {field}
+        </div>
+      )
+    }
+    else {
+      return (
+      <div key={i}>
+        <input name={field.key} type='text' placeholder={field.placeholder} value={inputFields[field.key]} onChange={onInputChange} />
```

Comment:
In the test, this form is creating these inputs with initial values `undefined` which is why you're getting the warning you are.  You can see in the test that the props are a bit different, and you could adjust the test.